### ISSUE TITLE
Peer dependencies issues when initializing new project with npm 7+

### ISSUE THREAD

**Issue by lebedev** [NomicFoundation/hardhat#4730]
### Version of Hardhat

2.19.4

### What happened?

I initialized a new project and tried to run example tests.

Expected results: all the required dependencies are automatically installed and tests run successfully from the get-go.

Actual results: I get an error with a broken link (leading to 404 page) about not installed peer dependencies and am required to install them manually. After that `npx hardhat test` fail all the tests with `TypeError: (0 , ethers_1.getAddress) is not a function` error text. Errors are gone after I additionally manually installed `ethers` package, which wasn't mentioned in HH801 error text.

### Minimal reproduction steps

I have Node@20/npm@10 (installed with `nvm`) on WSL2. Also tried with Node@18, and npm@7, and npm@9 with the same results. Here are the commands I used with console output.
```sh
angly@Nekst:~/hardhat-init-test$ node -v
v20.10.0
angly@Nekst:~/hardhat-init-test$ npm -v
10.2.3
```
An empty npm project is initialized.
```sh
angly@Nekst:~/hardhat-init-test$ npm init -y
Wrote to /home/angly/hardhat-init-test/package.json:

{
  "name": "hardhat-init-test",
  "version": "1.0.0",
  "description": "",
  "main": "index.js",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "keywords": [],
  "author": "",
  "license": "ISC"
}
```
Hardhat is installed.
```sh
angly@Nekst:~/hardhat-init-test$ npm i -D hardhat

added 312 packages, and audited 313 packages in 2s

69 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities
```
New hardhat JS project is initialized agreeing to all the options.
```sh
angly@Nekst:~/hardhat-init-test$ npx hardhat init
888    888                      888 888               888
888    888                      888 888               888
888    888                      888 888               888
8888888888  8888b.  888d888 .d88888 88888b.   8888b.  888888
888    888     "88b 888P"  d88" 888 888 "88b     "88b 888
888    888 .d888888 888    888  888 888  888 .d888888 888
888    888 888  888 888    Y88b 888 888  888 888  888 Y88b.
888    888 "Y888888 888     "Y88888 888  888 "Y888888  "Y888

Welcome to Hardhat v2.19.4

✔ What do you want to do? · Create a JavaScript project
✔ Hardhat project root: · /home/angly/hardhat-init-test
✔ Do you want to add a .gitignore? (Y/n) · y
✔ Do you want to install this sample project's dependencies with npm (@nomicfoundation/hardhat-toolbox)? (Y/n) · y

npm install --save-dev @nomicfoundation/hardhat-toolbox@^4.0.0

added 1 package, and audited 314 packages in 552ms

69 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities

Project created

See the README.md file for some example tasks you can run

Give Hardhat a star on Github if you're enjoying it!

     https://github.com/NomicFoundation/hardhat

Please take a moment to complete the 2023 Solidity Survey: https://hardhat.org/solidity-survey-2023
```
Tried to run tests, got unmet peer dependencies error with broken https://hardhat.org/HH801 link.
```sh
angly@Nekst:~/hardhat-init-test$ npx hardhat test
Error HH801: Plugin @nomicfoundation/hardhat-toolbox requires the following dependencies to be installed: @nomicfoundation/hardhat-chai-matchers, @nomicfoundation/hardhat-ethers, @nomicfoundation/hardhat-network-helpers, @nomicfoundation/hardhat-verify, @typechain/ethers-v6, @typechain/hardhat, @types/chai, @types/mocha, chai, hardhat-gas-reporter, solidity-coverage, ts-node, typechain, typescript.
Please run: npm install --save-dev "@nomicfoundation/hardhat-chai-matchers@^2.0.0" "@nomicfoundation/hardhat-ethers@^3.0.0" "@nomicfoundation/hardhat-network-helpers@^1.0.0" "@nomicfoundation/hardhat-verify@^2.0.0" "@typechain/ethers-v6@^0.5.0" "@typechain/hardhat@^9.0.0" "@types/chai@^4.2.0" "@types/mocha@>=9.1.0" "chai@^4.2.0" "hardhat-gas-reporter@^1.0.8" "solidity-coverage@^0.8.1" "ts-node@>=8.0.0" "typechain@^8.3.0" "typescript@>=4.5.0"

For more info go to https://hardhat.org/HH801 or run Hardhat with --show-stack-traces
```
Installed peer dependencies using the command from the output above.
```sh
angly@Nekst:~/hardhat-init-test$ npm install --save-dev "@nomicfoundation/hardhat-chai-matchers@^2.0.0" "@nomicfoundation/hardhat-ethers@^3.0.0" "@nomicfoundation/hardhat-network-helpers@^1.0.0" "@nomicfoundation/hardhat-verify@^2.0.0" "@typechain/ethers-v6@^0.5.0" "@typechain/hardhat@^9.0.0" "@types/chai@^4.2.0" "@types/mocha@>=9.1.0" "chai@^4.2.0" "hardhat-gas-reporter@^1.0.8" "solidity-coverage@^0.8.1" "ts-node@>=8.0.0" "typechain@^8.3.0" "typescript@>=4.5.0"

added 245 packages, and audited 559 packages in 23s

89 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities
```
Tried to run tests again, got a cryptic error message.
```sh
angly@Nekst:~/hardhat-init-test$ npx hardhat test
Compiled 1 Solidity file successfully (evm target: paris).


  Lock
    Deployment
      1) Should set the right unlockTime
      2) Should set the right owner
      3) Should receive and store the funds to lock
      4) Should fail if the unlockTime is not in the future
    Withdrawals
      Validations
        5) Should revert with the right error if called too soon
        6) Should revert with the right error if called from another account
        7) Shouldn't fail if the unlockTime has arrived and the owner calls it
      Events
        8) Should emit an event on withdrawals
      Transfers
        9) Should transfer the funds to the owner


  0 passing (431ms)
  9 failing

  1) Lock
       Deployment
         Should set the right unlockTime:
     TypeError: (0 , ethers_1.getAddress) is not a function
      at new HardhatEthersSigner (node_modules/@nomicfoundation/hardhat-ethers/src/signers.ts:73:30)
      at Function.create (node_modules/@nomicfoundation/hardhat-ethers/src/signers.ts:65:12)
      at processTicksAndRejections (node:internal/process/task_queues:95:5)
      at getSigner (node_modules/@nomicfoundation/hardhat-ethers/src/internal/helpers.ts:60:29)
      at async Promise.all (index 0)
      at getSigners (node_modules/@nomicfoundation/hardhat-ethers/src/internal/helpers.ts:45:30)
      at deployOneYearLockFixture (test/Lock.js:20:35)
      at loadFixture (node_modules/@nomicfoundation/hardhat-network-helpers/src/loadFixture.ts:59:18)
      at Context.<anonymous> (test/Lock.js:30:36)

  2) Lock
       Deployment
         Should set the right owner:
     TypeError: (0 , ethers_1.getAddress) is not a function
      at new HardhatEthersSigner (node_modules/@nomicfoundation/hardhat-ethers/src/signers.ts:73:30)
      at Function.create (node_modules/@nomicfoundation/hardhat-ethers/src/signers.ts:65:12)
      at processTicksAndRejections (node:internal/process/task_queues:95:5)
      at getSigner (node_modules/@nomicfoundation/hardhat-ethers/src/internal/helpers.ts:60:29)
      at async Promise.all (index 0)
      at getSigners (node_modules/@nomicfoundation/hardhat-ethers/src/internal/helpers.ts:45:30)
      at deployOneYearLockFixture (test/Lock.js:20:35)
      at loadFixture (node_modules/@nomicfoundation/hardhat-network-helpers/src/loadFixture.ts:59:18)
      at Context.<anonymous> (test/Lock.js:36:31)

  3) Lock
       Deployment
         Should receive and store the funds to lock:
     TypeError: (0 , ethers_1.getAddress) is not a function
      at new HardhatEthersSigner (node_modules/@nomicfoundation/hardhat-ethers/src/signers.ts:73:30)
      at Function.create (node_modules/@nomicfoundation/hardhat-ethers/src/signers.ts:65:12)
      at processTicksAndRejections (node:internal/process/task_queues:95:5)
      at getSigner (node_modules/@nomicfoundation/hardhat-ethers/src/internal/helpers.ts:60:29)
      at async Promise.all (index 0)
      at getSigners (node_modules/@nomicfoundation/hardhat-ethers/src/internal/helpers.ts:45:30)
      at deployOneYearLockFixture (test/Lock.js:20:35)
      at loadFixture (node_modules/@nomicfoundation/hardhat-network-helpers/src/loadFixture.ts:59:18)
      at Context.<anonymous> (test/Lock.js:42:38)

  4) Lock
       Deployment
         Should fail if the unlockTime is not in the future:
     TypeError: (0 , ethers_1.getAddress) is not a function
      at new HardhatEthersSigner (node_modules/@nomicfoundation/hardhat-ethers/src/signers.ts:73:30)
      at Function.create (node_modules/@nomicfoundation/hardhat-ethers/src/signers.ts:65:12)
      at processTicksAndRejections (node:internal/process/task_queues:95:5)
      at getSigner (node_modules/@nomicfoundation/hardhat-ethers/src/internal/helpers.ts:60:29)
      at async Promise.all (index 0)
      at getSigners (node_modules/@nomicfoundation/hardhat-ethers/src/internal/helpers.ts:45:30)
      at getContractFactoryByAbiAndBytecode (node_modules/@nomicfoundation/hardhat-ethers/src/internal/helpers.ts:305:21)
      at Context.<anonymous> (test/Lock.js:54:20)

  5) Lock
       Withdrawals
         Validations
           Should revert with the right error if called too soon:
     TypeError: (0 , ethers_1.getAddress) is not a function
      at new HardhatEthersSigner (node_modules/@nomicfoundation/hardhat-ethers/src/signers.ts:73:30)
      at Function.create (node_modules/@nomicfoundation/hardhat-ethers/src/signers.ts:65:12)
      at processTicksAndRejections (node:internal/process/task_queues:95:5)
      at getSigner (node_modules/@nomicfoundation/hardhat-ethers/src/internal/helpers.ts:60:29)
      at async Promise.all (index 0)
      at getSigners (node_modules/@nomicfoundation/hardhat-ethers/src/internal/helpers.ts:45:30)
      at deployOneYearLockFixture (test/Lock.js:20:35)
      at loadFixture (node_modules/@nomicfoundation/hardhat-network-helpers/src/loadFixture.ts:59:18)
      at Context.<anonymous> (test/Lock.js:64:26)

  6) Lock
       Withdrawals
         Validations
           Should revert with the right error if called from another account:
     TypeError: (0 , ethers_1.getAddress) is not a function
      at new HardhatEthersSigner (node_modules/@nomicfoundation/hardhat-ethers/src/signers.ts:73:30)
      at Function.create (node_modules/@nomicfoundation/hardhat-ethers/src/signers.ts:65:12)
      at processTicksAndRejections (node:internal/process/task_queues:95:5)
      at getSigner (node_modules/@nomicfoundation/hardhat-ethers/src/internal/helpers.ts:60:29)
      at async Promise.all (index 0)
      at getSigners (node_modules/@nomicfoundation/hardhat-ethers/src/internal/helpers.ts:45:30)
      at deployOneYearLockFixture (test/Lock.js:20:35)
      at loadFixture (node_modules/@nomicfoundation/hardhat-network-helpers/src/loadFixture.ts:59:18)
      at Context.<anonymous> (test/Lock.js:72:52)

  7) Lock
       Withdrawals
         Validations
           Shouldn't fail if the unlockTime has arrived and the owner calls it:
     TypeError: (0 , ethers_1.getAddress) is not a function
      at new HardhatEthersSigner (node_modules/@nomicfoundation/hardhat-ethers/src/signers.ts:73:30)
      at Function.create (node_modules/@nomicfoundation/hardhat-ethers/src/signers.ts:65:12)
      at processTicksAndRejections (node:internal/process/task_queues:95:5)
      at getSigner (node_modules/@nomicfoundation/hardhat-ethers/src/internal/helpers.ts:60:29)
      at async Promise.all (index 0)
      at getSigners (node_modules/@nomicfoundation/hardhat-ethers/src/internal/helpers.ts:45:30)
      at deployOneYearLockFixture (test/Lock.js:20:35)
      at loadFixture (node_modules/@nomicfoundation/hardhat-network-helpers/src/loadFixture.ts:59:18)
      at Context.<anonymous> (test/Lock.js:86:38)

  8) Lock
       Withdrawals
         Events
           Should emit an event on withdrawals:
     TypeError: (0 , ethers_1.getAddress) is not a function
      at new HardhatEthersSigner (node_modules/@nomicfoundation/hardhat-ethers/src/signers.ts:73:30)
      at Function.create (node_modules/@nomicfoundation/hardhat-ethers/src/signers.ts:65:12)
      at processTicksAndRejections (node:internal/process/task_queues:95:5)
      at getSigner (node_modules/@nomicfoundation/hardhat-ethers/src/internal/helpers.ts:60:29)
      at async Promise.all (index 0)
      at getSigners (node_modules/@nomicfoundation/hardhat-ethers/src/internal/helpers.ts:45:30)
      at deployOneYearLockFixture (test/Lock.js:20:35)
      at loadFixture (node_modules/@nomicfoundation/hardhat-network-helpers/src/loadFixture.ts:59:18)
      at Context.<anonymous> (test/Lock.js:99:52)

  9) Lock
       Withdrawals
         Transfers
           Should transfer the funds to the owner:
     TypeError: (0 , ethers_1.getAddress) is not a function
      at new HardhatEthersSigner (node_modules/@nomicfoundation/hardhat-ethers/src/signers.ts:73:30)
      at Function.create (node_modules/@nomicfoundation/hardhat-ethers/src/signers.ts:65:12)
      at processTicksAndRejections (node:internal/process/task_queues:95:5)
      at getSigner (node_modules/@nomicfoundation/hardhat-ethers/src/internal/helpers.ts:60:29)
      at async Promise.all (index 0)
      at getSigners (node_modules/@nomicfoundation/hardhat-ethers/src/internal/helpers.ts:45:30)
      at deployOneYearLockFixture (test/Lock.js:20:35)
      at loadFixture (node_modules/@nomicfoundation/hardhat-network-helpers/src/loadFixture.ts:59:18)
      at Context.<anonymous> (test/Lock.js:113:59)




Please take a moment to complete the 2023 Solidity Survey: https://hardhat.org/solidity-survey-2023
```
Then I thought of comparing the list of peer dependencies from HH801 error with the list from the docs and discovered that `ethers` was missing in HH801 error's text and installed it manually. After that tests run successfully.

Please help me understand how to make peer dependencies to be installed automatically.

### Search terms

init peer dependencies

**Comment by fvictorio** [comment]
@lebedev can you try this in an empty directory and tell me what happens?

```
npm init -y
npm install --save-dev @nomicfoundation/hardhat-toolbox
```

This should auto-install the toolbox peer dependencies, so instead of "added 1 package" you should get way more.

One possibility here is that (for some reason, perhaps nvm-related), the auto-installation done by Hardhat is using a different version of npm (maybe the sytem version), and that version doesn't auto-install peer deps. But maybe something else is going on here.

**Comment by lebedev** [comment]
I solved it!

I wondered why there wasn't any warnings from npm about missing peer dependencies and why my npm 7+ didn't install any peer dependencies when it by all means should have. Turned out I had `legacy-peer-deps` option set (which exists exactly for disabling peer dependencies installation) and completely forgot about it. This option can be provided as a flag to `npm i`, as a key-value pair in a `.npmrc` file, and as a `npm config` option (which was my case), and this option surprisingly persisted node versions uninstallations/reinstallations via `nvm`.

Thanks for you assistance, @fvictorio!
